lib/HTTP/Response.pm

```diff
@@ -126,18 +126,36 @@ sub filename
     if ($cd) {
 	require HTTP::Headers::Util;
 	if (my @cd = HTTP::Headers::Util::split_header_words($cd)) {
-	    my ($disposition, undef, %cd_param) = @{$cd[-1]};
-	    $file = $cd_param{filename};
-
-	    # RFC 2047 encoded?
-	    if ($file && $file =~ /^=\?(.+?)\?(.+?)\?(.+)\?=$/) {
-		my $charset = $1;
-		my $encoding = uc($2);
-		my $encfile = $3;
-
+		my ($disposition, undef, %cd_param) = @{$cd[-1]};
+
+		$file = $cd_param{filename};
+
+		my ($charset, $encoding, $lang, $encfile);
+
+		if( exists $cd_param{'filename*'} &&
```

Comment:
Could you run perltidy on this, please?

```bash
$ perltidy lib/HTTP/Response.pm -o clean_response.pm
```
open clean_response.pm, copy the clean code from the new function and any other portion you've updated and overwrite that section in lib/HTTP/Response.pm with the new, cleaned up version.

```bash
$ rm clean_response.pm
$ 
```

Reply:
I was wondering about that. The indentation was inconsistent for the older code in the function and it was bothering me.